lib/webidl2.js

```diff
@@ -1,29 +1,69 @@
 "use strict";
 
 (() => {
+  // These regular expressions use the sticky flag so they will only match at
+  // the current location (ie. the offset of lastIndex).
+  const tokenRe = {
+    // This expression uses a lookahead assertion to catch false matches
+    // against integers early.
+    "float": /-?(?=[0-9]*\.|[0-9]+[eE])(([0-9]+\.[0-9]*|[0-9]*\.[0-9]+)([Ee][-+]?[0-9]+)?|[0-9]+[Ee][-+]?[0-9]+)/y,
```

Comment:
I want to note that this is now branching from [the spec](https://heycam.github.io/webidl/#idl-grammar). Maybe worth pushing to upstream if this gives a significant win.

Reply:
I think it would be not be a good match for the spec because it makes the expression harder to read and understand. It offers a significant performance benefit in this tokeniser, because of the way we run the expression against every digit, most of which will fail the lookahead assertion. But it might just get in the way in a different type of tokeniser.